### ISSUE TITLE
No stats breakdown of numbers on patient level histogram

### ISSUE THREAD

**Issue by bsmarine** [bsmarine/BleedDetection#4]
When plotting a histogram for patient level statistics, there is no labels showing number of true positives, false positives, true negatives and false negatives. This likely is due to how patient-level results are inputted differently into plotting code at evaluator.py line 275.

Would it be possible to include these statistics somehow?

**Comment by Sonic81518** [comment]
After running evaluator.py, it will generate a Dataframe with the prediction on each box (see the image below).
We could generate another Dataframe from that, and each row represents a unique patient with predictive confidence. Then you can use a threshold (maybe 0.5) to turn the predictive confidence into binary prediction.
![original Dataframe](https://user-images.githubusercontent.com/11848411/81545403-6b78f000-93ab-11ea-9d24-9a9aa5afa3ef.png)

**Comment by bsmarine** [comment]
ROI-level average precision [true positives/(true positives + false positives)] is calculated already for summary stats using a minimum threshold confidence, but only for positive cases. This is seen in evaluator.py line 465. 

Patient-level stats (AUC) currently are calculated using the highest confidence ROI for a patient, which is okay. This is done in evaluator.py line 293. But the summary does not include number of TN, FN, TP, FP which would be helpful.

Another question: there are stats such as in row 13155 above with det_type of det_fn... where do negative predictions come from for object detection?

**Comment by hankchen1728** [comment]
I modified the `evaluator.py` to make it preserve the **TP, FP, FN** informations. The figure below is the patient-level histogram, now it contain the **TP, FP, FN** informations in the title.
This modification will be pushed to a new branch later.
![pred_hist_hold_out_test_patient_cl1](https://user-images.githubusercontent.com/24695822/82749328-0d043680-9ddb-11ea-9f40-bd16b7c4e55d.png)

For the question about `det_fn`, the objects with `det_fn` type are just ground truth where no candidates match them. This information was added to the dataframe for processing convenience.
